sdcm/nemesis.py

```diff
@@ -2141,6 +2141,26 @@ def delete_half_partition(self, ks_cf):
 
         return partitions_for_delete
 
+    def delete_range_with_timestamp(self, ks_cf):
+        self.log.debug('Delete single partition range with timestamp')
+        partition_to_delete = self.choose_partitions_for_delete(partitions_amount=1, ks_cf=ks_cf)
+        if not partition_to_delete:
+            self.log.error('Not found partitions for delete')
+            return
+
+        pk_to_use = partition_to_delete.keys()[0]
+
+        with self.cluster.cql_connection_patient() as session:
+            full_partition = session.execute(f"select pk, ck from {ks_cf} where pk = {pk_to_use}").all()
+            delete_mark = int(len(full_partition) * 0.25)
+            timestamp = session.execute(
+                f"select pk, ck from {ks_cf} where pk = {pk_to_use} and ck = {delete_mark}").one().writetime_y
+            if not timestamp:
+                self.log.error(f"Unable to get writetime for row {pk_to_use},{delete_mark}")
+                return
+
+            session.execute(f"delete from {ks_cf} where pk = {pk_to_use} using timestamp {timestamp}")
```

Comment:
missing verification - e.g. count rows before and after deletion - verify how many deleted.